<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="simphony">
<title>Using simphony to evaluate rhythm detection • simphony</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using simphony to evaluate rhythm detection">
<meta property="og:description" content="simphony">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">simphony</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/examples.html">Using simphony's various options</a>
    <a class="dropdown-item" href="../articles/introduction.html">Using simphony to evaluate rhythm detection</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/hugheylab/simphony/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="introduction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="introduction_files/kePrint-0.0.1/kePrint.js"></script><link href="introduction_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Using simphony to evaluate rhythm detection</h1>
            
            <h4 data-toc-skip class="date">2022-05-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/hugheylab/simphony/blob/HEAD/vignettes/introduction.Rmd" class="external-link"><code>vignettes/introduction.Rmd</code></a></small>
      <div class="d-none name"><code>introduction.Rmd</code></div>
    </div>

    
    
<p><code>simphony</code> is a framework for simulating rhythmic data, especially gene expression data. Here we show an example of using it to benchmark a method for detecting rhythmicity.</p>
<div class="section level2">
<h2 id="load-the-packages-well-use">Load the packages we’ll use<a class="anchor" aria-label="anchor" href="#load-the-packages-well-use"></a>
</h2>
<p>Internally, <code>simphony</code> uses the <code>data.table</code> package, which provides an enhanced version of the standard R <code>data.frame</code>. We’ll use <code>data.table</code> for this example as well.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://r-datatable.com" class="external-link">'data.table'</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://haozhu233.github.io/kableExtra/" class="external-link">'kableExtra'</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">'knitr'</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">'limma'</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/evalclass/precrec" class="external-link">'precrec'</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://simphony.hugheylab.org">'simphony'</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="simulate-the-data">Simulate the data<a class="anchor" aria-label="anchor" href="#simulate-the-data"></a>
</h2>
<p>Here we create a <code>data.table</code> called <code>featureGroups</code> that specifies the desired properties of the simulated genes. We want 75% of simulated genes to be non-rhythmic and 25% to have a rhythm amplitude of 1.1. Properties not specified in <code>featureGroups</code> will be given their default values.</p>
<p>Our simulated experiment will have 200 genes. Expression values will be sampled from the negative binomial family, which models read counts from next-generation sequencing data. The interval between time points will be 2 (default period of 24), with one replicate per time point. We also use the default time range of our simulated data points of between 0 and 48 hours.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">44</span><span class="op">)</span>
<span class="va">featureGroups</span> <span class="op">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span>fracFeatures <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.75</span>, <span class="fl">0.25</span><span class="op">)</span>, amp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.3</span><span class="op">)</span><span class="op">)</span>
<span class="va">simData</span> <span class="op">=</span> <span class="fu"><a href="../reference/simphony.html">simphony</a></span><span class="op">(</span><span class="va">featureGroups</span>, nFeatures <span class="op">=</span> <span class="fl">200</span>, interval <span class="op">=</span> <span class="fl">2</span>, nReps <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">'negbinom'</span><span class="op">)</span></code></pre></div>
<p>The output of <code>simphony</code> has three components: <code>abundData</code>, <code>sampleMetadata</code>, and <code>featureMetadata</code>. <code>abundData</code> is a matrix that contains the simulated expression values. Each row of corresponds to a gene, each column corresponds to a sample. Since we sampled from the negative binomial family, all expression values are integers.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">simData</span><span class="op">$</span><span class="va">abundData</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
sample_01
</th>
<th style="text-align:right;">
sample_02
</th>
<th style="text-align:right;">
sample_03
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
feature_001
</td>
<td style="text-align:right;">
282
</td>
<td style="text-align:right;">
154
</td>
<td style="text-align:right;">
181
</td>
</tr>
<tr>
<td style="text-align:left;">
feature_002
</td>
<td style="text-align:right;">
213
</td>
<td style="text-align:right;">
310
</td>
<td style="text-align:right;">
200
</td>
</tr>
<tr>
<td style="text-align:left;">
feature_003
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
294
</td>
<td style="text-align:right;">
273
</td>
</tr>
</tbody>
</table>
<p><code>sampleMetadata</code> is a <code>data.table</code> that contains the condition (<code>cond</code>) and time for each sample. Here we simulated one condition, so <code>cond</code> is 1 for all samples.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">simData</span><span class="op">$</span><span class="va">sampleMetadata</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
sample
</th>
<th style="text-align:left;">
cond
</th>
<th style="text-align:right;">
time
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
sample_01
</td>
<td style="text-align:left;">
cond_1
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
sample_02
</td>
<td style="text-align:left;">
cond_1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
sample_03
</td>
<td style="text-align:left;">
cond_1
</td>
<td style="text-align:right;">
4
</td>
</tr>
</tbody>
</table>
<p><code>featureMetadata</code> is a <code>data.table</code> that contains the properties of each simulated gene in each condition. The <code>group</code> column corresponds to the row in <code>featureGroups</code> to which the gene belongs.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">simData</span><span class="op">$</span><span class="va">featureMetadata</span><span class="op">[</span><span class="fl">149</span><span class="op">:</span><span class="fl">151</span>, <span class="op">!</span><span class="st">'dispFunc'</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span>font_size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span></code></pre></div>
<table class="table table" style="font-size: 12px; margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
cond
</th>
<th style="text-align:right;">
group
</th>
<th style="text-align:left;">
feature
</th>
<th style="text-align:right;">
fracFeatures
</th>
<th style="text-align:left;">
amp
</th>
<th style="text-align:right;">
amp0
</th>
<th style="text-align:right;">
phase
</th>
<th style="text-align:right;">
period
</th>
<th style="text-align:left;">
rhyFunc
</th>
<th style="text-align:left;">
base
</th>
<th style="text-align:right;">
base0
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
cond_1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
feature_149
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:left;">
function (m) , x
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:left;">
.Primitive(“sin”)
</td>
<td style="text-align:left;">
function (x) , defaultValue
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
cond_1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
feature_150
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:left;">
function (m) , x
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:left;">
.Primitive(“sin”)
</td>
<td style="text-align:left;">
function (x) , defaultValue
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
cond_1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
feature_151
</td>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:left;">
function (m) , x
</td>
<td style="text-align:right;">
0.3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:left;">
.Primitive(“sin”)
</td>
<td style="text-align:left;">
function (x) , defaultValue
</td>
<td style="text-align:right;">
8
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="plot-the-simulated-time-course-for-selected-genes">Plot the simulated time-course for selected genes<a class="anchor" aria-label="anchor" href="#plot-the-simulated-time-course-for-selected-genes"></a>
</h2>
<p>Here we plot the simulated time-course for a non-rhythmic gene and a rhythmic gene. We use the <code>mergeSimData</code> function to merge the expression values, the sample metadata, and the gene metadata.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fmExample</span> <span class="op">=</span> <span class="va">simData</span><span class="op">$</span><span class="va">featureMetadata</span><span class="op">[</span><span class="va">feature</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'feature_150'</span>, <span class="st">'feature_151'</span><span class="op">)</span><span class="op">]</span>
<span class="va">dExample</span> <span class="op">=</span> <span class="fu"><a href="../reference/mergeSimData.html">mergeSimData</a></span><span class="op">(</span><span class="va">simData</span>, <span class="va">fmExample</span><span class="op">$</span><span class="va">feature</span><span class="op">)</span></code></pre></div>
<p>We also want to compare the simulated expression values with their underlying distributions over time, for which we can use the <code>getExpectedAbund</code> function. Since we sampled from the negative binomial family, the resulting <code>mu</code> column corresponds to the expected log<sub>2</sub> counts.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dExpect</span> <span class="op">=</span> <span class="fu"><a href="../reference/getExpectedAbund.html">getExpectedAbund</a></span><span class="op">(</span><span class="va">fmExample</span>, <span class="fl">24</span>, times <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">48</span>, <span class="fl">0.25</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Then it all comes together with <code>ggplot</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dExample</span><span class="op">[</span>, <span class="va">featureLabel</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">feature</span>, <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">amp0</span> <span class="op">==</span> <span class="fl">0</span>, <span class="st">'(non-rhythmic)'</span>, <span class="st">'(rhythmic)'</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="va">dExpect</span><span class="op">[</span>, <span class="va">featureLabel</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">feature</span>, <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">amp0</span> <span class="op">==</span> <span class="fl">0</span>, <span class="st">'(non-rhythmic)'</span>, <span class="st">'(rhythmic)'</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dExample</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">featureLabel</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">time</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">2</span><span class="op">^</span><span class="va">mu</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.25</span>, data <span class="op">=</span> <span class="va">dExpect</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">time</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">abund</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fl">21</span>, size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Time (h)'</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="va">log</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">(</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">48</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">48</span>, <span class="fl">8</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-9-1.png" width="576"></p>
</div>
<div class="section level2">
<h2 id="detect-rhythmic-genes">Detect rhythmic genes<a class="anchor" aria-label="anchor" href="#detect-rhythmic-genes"></a>
</h2>
<p>We can use the <code>limma</code> package to detect rhythmic genes based on a linear model that corresponds to cosinor regression.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sampleMetadata</span> <span class="op">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/copy.html" class="external-link">copy</a></span><span class="op">(</span><span class="va">simData</span><span class="op">$</span><span class="va">sampleMetadata</span><span class="op">)</span>
<span class="va">sampleMetadata</span><span class="op">[</span>, <span class="va">timeCos</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/Trig.html" class="external-link">cos</a></span><span class="op">(</span><span class="va">time</span> <span class="op">*</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">pi</span> <span class="op">/</span> <span class="fl">24</span><span class="op">)</span><span class="op">]</span>
<span class="va">sampleMetadata</span><span class="op">[</span>, <span class="va">timeSin</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/Trig.html" class="external-link">sin</a></span><span class="op">(</span><span class="va">time</span> <span class="op">*</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">pi</span> <span class="op">/</span> <span class="fl">24</span><span class="op">)</span><span class="op">]</span>
<span class="va">design</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">timeCos</span> <span class="op">+</span> <span class="va">timeSin</span>, data <span class="op">=</span> <span class="va">sampleMetadata</span><span class="op">)</span></code></pre></div>
<p>Here we follow the typical <code>limma</code> workflow: fit the linear model for each gene, run empirical Bayes, and extract the relevant summary statistics. We pass <code>lmFit</code> the log<sub>2</sub> transformed counts.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">simData</span><span class="op">$</span><span class="va">abundData</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span>, trend <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">rhyLimma</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, number <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="evaluate-accuracy-of-rhythmic-gene-detection">Evaluate accuracy of rhythmic gene detection<a class="anchor" aria-label="anchor" href="#evaluate-accuracy-of-rhythmic-gene-detection"></a>
</h2>
<p>First we merge the results from <code>limma</code> with the known amplitudes from <code>featureMetadata</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rhyLimma</span><span class="op">$</span><span class="va">feature</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rhyLimma</span><span class="op">)</span>
<span class="va">rhyLimma</span> <span class="op">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span><span class="va">rhyLimma</span><span class="op">)</span>, <span class="va">simData</span><span class="op">$</span><span class="va">featureMetadata</span><span class="op">[</span>, <span class="fu">.</span><span class="op">(</span><span class="va">feature</span>, <span class="va">amp0</span><span class="op">)</span><span class="op">]</span>, by <span class="op">=</span> <span class="st">'feature'</span><span class="op">)</span></code></pre></div>
<p>We can plot the distributions of p-values of rhythmicity for non-rhythmic and rhythmic genes. P-values for non-rhythmic genes are uniformly distributed between 0 and 1, as they should be under the null hypothesis. P-values for rhythmic genes, on the other hand, tend to be closer to 0.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">rhyLimma</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">amp0</span><span class="op">)</span>, y <span class="op">=</span> <span class="va">P.Value</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fl">21</span>, width <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="st">'Rhythm amplitude '</span> <span class="op">*</span> <span class="op">(</span><span class="va">log</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">~</span> <span class="va">counts</span><span class="op">)</span><span class="op">)</span>, y <span class="op">=</span> <span class="st">'P-value of rhythmicity'</span><span class="op">)</span></code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-13-1.png" width="336"></p>
<p>Finally, we can summarize the ability to distinguish non-rhythmic and rhythmic genes using a receiver operating characteristic (ROC) curve (here we use the <code>precrec</code> package).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rocprc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/precrec/man/evalmod.html" class="external-link">evalmod</a></span><span class="op">(</span>scores <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">rhyLimma</span><span class="op">$</span><span class="va">P.Value</span><span class="op">)</span>, labels <span class="op">=</span> <span class="va">rhyLimma</span><span class="op">$</span><span class="va">amp0</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">rocprc</span>, <span class="st">'ROC'</span><span class="op">)</span></code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-14-1.png" width="288"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jake Hughey, Jordan Singer.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
